shared/lib/wp/support.js

```diff
@@ -0,0 +1,102 @@
+export default function wpcomSupport( wpcom ) {
+	let supportUser = '';
+	let supportToken = '';
+
+	/**
+	 * Request parameters are as follows:
+	 * - param (required) A string or object that contains the path
+	 * - query (optional) An object that expands the response object
+	 * - body (required*) Required by POST and PUT, but not by GET and DEL
+	 * - fn (required) A callback function to handle the returned result
+	 * - ...rest (optional) Some queries have additional parameters after
+	 *   callback function
+	 *
+	 * Return the index of the query parameter, or if one does not exist,
+	 * return false.
+	 */
+	const getQueryIndex = function( req, args ) {
+		let fnIndex, queryIndex;
+
+		// Find the index of the callback in the arguments
+		fnIndex = args.findIndex( function( e ) {
```

Comment:
Maybe use the lodash equivalent here. Not sure this is polyfilled currently. I'll check during testing.